firefox-ios/Client/Application/RemoteSettingsUtil.swift

```diff
@@ -0,0 +1,248 @@
+            }
+        }
+    }
+
+    func saveRemoteSettingsRecord(_ records: [RemoteSettingsRecord], forKey key: String) {
+        let encoder = JSONEncoder()
+        if let encoded = try? encoder.encode(records) {
+            UserDefaults.standard.set(encoded, forKey: key)
+        }
+    }
+
+    func fetchLocalRecords(forKey key: String) -> [RemoteSettingsRecord]? {
+        if let savedData = UserDefaults.standard.data(forKey: key) {
+            let decoder = JSONDecoder()
+            if let loadedRecords = try? decoder.decode([RemoteSettingsRecord].self, from: savedData) {
+                return loadedRecords
+            }
+        }
+        return nil
+    }
+
+    func fetchRemoteRecords(collectionName: RemoteCollection,
+                              completion: @escaping (Result<[RemoteSettingsRecord], RemoteSettingsUtilError>) -> Void) {
+        do {
+            let config = RemoteSettingsConfig(collectionName: collectionName.rawValue)
+            let settings = try RemoteSettings(remoteSettingsConfig: config)
+            let response = try settings.getRecords()
+            completion(.success(response.records))
+        } catch {
+            completion(.failure(.fetchError(error)))
+        }
+    }
+
+    func areRecordsDifferent(localRecords: [RemoteSettingsRecord],
+                             remoteRecords: [RemoteSettingsRecord]) -> Bool {
+        guard localRecords.count == remoteRecords.count else {
+            return true
+        }
+        for (local, remote) in zip(localRecords, remoteRecords) {
+            if local.lastModified != remote.lastModified {
+                return true
+            }
+        }
+        return false
+    }
+
+    func updateAndFetchRecords(for collectionName: RemoteCollection,
+                               completion: @escaping (Result<[RemoteSettingsRecord], Error>) -> Void) {
+        let localRecords = fetchLocalRecords(forKey: PrefsKeys.remoteSettingsKey) ?? []
+        
+        fetchRemoteRecords(collectionName: collectionName) { [weak self] result in
+            switch result {
+            case .success(let remoteRecords):
+                if self?.areRecordsDifferent(localRecords: localRecords,
+                                             remoteRecords: remoteRecords) ?? false {
+                    self?.saveRemoteSettingsRecord(remoteRecords, forKey: PrefsKeys.remoteSettingsKey)
+                    completion(.success(remoteRecords))
+                } else {
+                    completion(.success(localRecords))
+                }
+            case .failure(let error):
+                self?.logger.log("Failed to get any events from Remote Settings",
+                                       level: .warning,
+                                       category: .remoteSettings,
+                                       description: error.localizedDescription)
+                completion(.failure(error))
+            }
+        }
+    }
```

Comment:
Not a fan of this method either but this falls under what should we expect out of remote settings, should we be looking at the records and check if they are different via `lastModified` date in `areRecordsDifferent` or how else would we know if we are being served new vs stale content

Reply:
I think [get_records_since](https://github.com/mozilla/application-services/blob/71578bea70b5f9da0b9c35c17e8fe15e9cf2ebc9/components/remote_settings/src/remote_settings.udl#L70) is the way to do this.

Reply:
@bendk if we do that then records are still not cached so in theory does it make sense to cache it by using `fetchLocalRecords(forKey: PrefsKeys.remoteSettingsKey) ?? []`